packages/server/src/api/controllers/model.js

```diff
@@ -18,24 +18,25 @@ exports.find = async function(ctx) {
 
 exports.create = async function(ctx) {
   const db = new CouchDB(ctx.user.instanceId)
-  const newModel = {
+  const modelToSave = {
     type: "model",
-    ...ctx.request.body,
     _id: newid(),
+    ...ctx.request.body,
   }
 
-  const result = await db.post(newModel)
-  newModel._rev = result.rev
+  const result = await db.post(modelToSave)
+  modelToSave._rev = result.rev
 
   const { schema } = ctx.request.body
   for (let key in schema) {
     // model has a linked record
     if (schema[key].type === "link") {
       // create the link field in the other model
       const linkedModel = await db.get(schema[key].modelId)
-      linkedModel.schema[newModel.name] = {
+      linkedModel.schema[modelToSave.name] = {
```

Comment:
This suggests that a model can only link to another once ?
 e.g. is the following possible...

```
{
  "Primary Contact" : (link to Contact model),
  "Secondary Contact" : (link to contact model)
}
```

Also, this method does not withstand renaming of a model.

suggest, rather than `linkedModel.schema[newModel.name]` ... we do `linkedModel.schema[newModel._id + "." + key` ... and translate the `_id.key` to `name key` at runtime... e.g. "Customer, Primary Contact"

Reply:
Yep, my understanding is that we were doing something like airtable where you could only link to another once. We still shouldn't use names though due to the renaming issue